concepts/admission-webhook-cert-management/README.md

```diff
@@ -0,0 +1,145 @@
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: validation.webhook.telemetry.kyma-project.io
+webhooks:
+- clientConfig:
+    caBundle: {{ b64enc $ca.Cert }}
+...
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: telemetry-webhook-cert
+  labels:
+    {{- include "operator.labels" . | nindent 4 }}
+    {{- toYaml .Values.extraLabels | nindent 4 }}
+type: Opaque
+data:
+  tls.crt: {{ b64enc $cert.Cert }}
+  tls.key: {{ b64enc $cert.Key }}
+```
+
+This is a very simple solution, but it has a lot of disadvantages.
+The certs are updated every time the Helm chart is rendered (every reconciliation). This update is not atomic. For example, when the server cert is updated, but the caBundle of the webhook configuration is not yet updated, the webhook is in a non-working state, and all the corresponding API requests fail. This situation is rather common and is documented in [bug #15142](https://github.com/kyma-project/kyma/issues/15142). 
+
+You can come up with some workarounds: make reconciler deploy resources in a strict predefined order or make sure the webhook chart does not contain the corresponding CRs. However, it does not fix the underlying problem and the bug may pop up again. 
+
+
+### <a name="server-code"></a>Generate the CA cert and the server cert and update the webhook configuration in the webhook server code itself
+
+This approach is used by serverless and api-gateway. In this case, the certificates are generated upon the server startup. Here's an [example of this solution](https://github.com/kyma-project/api-gateway/blob/main/internal/webhook/certificates.go).
+
+Issuing a certificate in the webhook server code doesn't have the aforementioned problem. However, in this case it must be implemented by every operator. In addition to that, the webhook server must be provided with extended permissions to change the corresponding `validatingwebhookconfiguration` (or the `mutatingwebhookconfiguration`). Furthermore, both the CA and the server certificate are recreated upon each Pod restart and are never be rotated if the Pod is not restarted.
+
+
+## How other organizations manage webhook certificates
+
+The most prominent operator frameworks advertise `cert-manager` as a possible solution to manage webhook certificate:
+
+- Kubebuilder: https://kubebuilder.io/cronjob-tutorial/cert-manager.html
+- Operator SDK: https://developer.ibm.com/tutorials/create-a-conversion-webhook-with-the-operator-sdk/
+
+Let's discribe a few possible solutions.
+
+### <a name="gardener-cert-manager"></a>Gardener cert-manager
+
+https://github.com/gardener/cert-management
+
+Since Kyma-on-Gardener is a very common setup, and Gardener has a pre-installed cert-manager, it seems to be a natural choice. However Gardener `cert-manager` is actually a separate open source project and is very different from the upstrem `cert-manager`. It comes with a different feature set (it even has its own CRDs). From the Github project description:
```

Comment:
```suggestion
Because Kyma-on-Gardener is a very common setup, and Gardener has a pre-installed cert-manager, it seems to be a natural choice. However, Gardener `cert-manager` is actually a separate open source project and is very different from the upstream `cert-manager`. It comes with a different feature set (it even has its own CRDs). From the Github project description:
```